Tools/_Starlight/partition_tests.py

```
@@ -383,16 +496,64 @@ def cmd_read():
+
+    trx_files = glob.glob(os.path.join(trx_dir, "**", "*.trx"), recursive=True)
+    if not trx_files:
+        print(f"Error: no .trx files found under {trx_dir}", file=sys.stderr)
+        sys.exit(1)
+
+    totals = {}
+    counts = {}
+    parsed = 0
+    for path in trx_files:
+        try:
+            root = ET.parse(path).getroot()
```

Ruff (0.16.0)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)